src/mono/mono/mini/interp/transform-simd.c

```diff
@@ -192,8 +209,31 @@ resolve_native_size (MonoTypeEnum type)
 		return MONO_TYPE_U8;
 #endif
 	return type;
+}
+
+static const char *
+strip_explicit_isimd_prefix (const char *cmethod_name)
+{
+	if (strncmp(cmethod_name, "System.Runtime.Intrinsics.ISimdVector<System.", 45) == 0) {
+		if (strncmp (cmethod_name + 45, "Runtime.Intrinsics.Vector", 25) == 0) {
+		// We want explicitly implemented ISimdVector<TSelf, T> APIs to still be expanded where possible
+		// but, they all prefix the qualified name of the interface first, so we'll check for that and
+		// skip the prefix before trying to resolve the method.
 
+			if (strncmp(cmethod_name + 70, "64<T>,T>.", 9) == 0) {
+				cmethod_name += 79;
+			} else if ((strncmp(cmethod_name + 70, "128<T>,T>.", 10) == 0) ||
+				(strncmp(cmethod_name + 70, "256<T>,T>.", 10) == 0) ||
+				(strncmp(cmethod_name + 70, "512<T>,T>.", 10) == 0)) {
+				cmethod_name += 80;
+			}
```

Comment:
shouldn't we strip only 128bit version names here?

Reply:
It shouldn't matter in practice because we'll only reach this code in those cases but I can remove the extra cases